src/client/components/auto-refresh-menu/auto-refresh-menu.tsx

```diff
@@ -50,43 +52,47 @@ export interface AutoRefreshMenuProps {
   openOn: Element;
   onClose: Fn;
   autoRefreshRate: Duration;
-  setAutoRefreshRate: (duration: Duration) => void;
+  setAutoRefreshRate: Unary<Duration, void>;
   refreshMaxTime: Fn;
   dataCube: DataCube;
   timekeeper: Timekeeper;
   timezone: Timezone;
 }
 
-export interface AutoRefreshMenuState {
-}
-
-export class AutoRefreshMenu extends React.Component<AutoRefreshMenuProps, AutoRefreshMenuState> {
-  renderRefreshIntervalDropdown() {
-    const { autoRefreshRate, setAutoRefreshRate } = this.props;
-
-    return <Dropdown<Duration>
-      label={STRINGS.autoUpdate}
-      items={REFRESH_DURATIONS}
-      selectedItem={autoRefreshRate}
-      renderItem={d => AUTO_REFRESH_LABELS[String(d)] || `Custom ${d}`}
-      onSelect={setAutoRefreshRate}
-    />;
-  }
+const STAGE = Stage.fromSize(240, 200);
 
-  render() {
-    const { openOn, onClose, dataCube, timekeeper, timezone } = this.props;
-    const stage = Stage.fromSize(240, 200);
+function renderRefreshIntervalDropdown(autoRefreshRate: Duration, setAutoRefreshRate: Unary<Duration, void>) {
+  return <Dropdown<Duration>
+    label={STRINGS.autoUpdate}
+    items={REFRESH_DURATIONS}
+    selectedItem={autoRefreshRate}
+    renderItem={d => AUTO_REFRESH_LABELS[String(d)] || `Custom ${d}`}
+    onSelect={setAutoRefreshRate}
+  />;
+}
 
-    return <BubbleMenu
-      className="auto-refresh-menu"
-      direction="down"
-      stage={stage}
-      openOn={openOn}
-      onClose={onClose}
-    >
-      {this.renderRefreshIntervalDropdown()}
-      <button className="update-now-button" onClick={this.props.refreshMaxTime}>Update now</button>
-      <div className="update-info">{dataCube.updatedText(timekeeper, timezone)}</div>
-    </BubbleMenu>;
+function updatedText(dataCube: DataCube, timekeeper: Timekeeper, timezone: Timezone): string {
+  const { refreshRule } = dataCube;
+  if (refreshRule.isRealtime()) {
+    return "Updated ~1 second ago";
+  } else if (refreshRule.isFixed()) {
+    return `Fixed to ${getWallTimeString(refreshRule.time, timezone)}`;
+  } else { // refreshRule is query
+    const maxTime = dataCube.getMaxTime(timekeeper);
+    if (!maxTime) return null;
+    return `Updated ${getRelativeTime(maxTime, timezone)} ago`;
```

Comment:
I would expect getRelativeTimeString function name, or getWallTime instead of getWallTimeString